### PR TITLE
Filter out not application targets

### LINKED ISSUE #30: Exclude unit tests target from carting update action
When I launch `carting update` on my project it creates a new "Carthage" build phase for unit tests target too. The Carthage tutorial says that `copy-frameworks` action is not needed for unit tests and it's very annoyable to deleting this script each time. Or maybe I doing something wrong?

### PR DESCRIPTION
**Pull request by alphatroya**
Close #30

### CODE REVIEW

File: Sources/CartingCore/Models/TargetBody.swift [hunk 1]
```diff
@@ -53,3 +53,23 @@ final class TargetBody {
         self.productType = productType
     }
 }
+
+struct ProductType {
```
Comment (artemnovichkov) [inline Sources/CartingCore/Models/TargetBody.swift]: Please, move the struct to different file

File: Sources/CartingCore/Models/TargetBody.swift [hunk 2]
```diff
@@ -53,3 +53,23 @@ final class TargetBody {
...
+    private let value: String
+    init(_ value: String) {
```
Comment (artemnovichkov) [inline Sources/CartingCore/Models/TargetBody.swift]: According to Swift Style Guide:
```suggestion
    init(value: String) {
```

**Review by artemnovichkov — CHANGES_REQUESTED** [review]

**Comment by artemnovichkov** [comment]
Why didn't you add an enum? For instance:
```swift
enum ProductType {
    case application
    case other
}
```
I guess it's more flexible and will give a possibility to extend with different checks

**Comment by alphatroya** [comment]
> Why didn't you add an enum? For instance:
> 
> ```swift
> enum ProductType {
>     case application
>     case other
> }
> ```
> 
> I guess it's more flexible and will give a possibility to extend with different checks

I think about it. But other case should store value variable for `description` property hence we should throw up `String` inheritance and blow up implementation

``` swift
enum ProductType {

    case application
    case other(value: String)

    private static let applicationRawValue = "\"com.apple.product-type.application\""

    init(value: String) {
        switch value {
        case ProductType.applicationRawValue:
            self = .application
        default:
            self = .other(value: value)
        }
    }

    var description: String {
        switch self {
        case .application:
            return ProductType.applicationRawValue
        case let .other(value):
            return value
        }
    }

    var isApplication: Bool {
        switch self {
        case .application:
            return true
        default:
            return false
        }
    }
}
```

**Comment by artemnovichkov** [comment]
I agree, good point! The last comment - could we format `"\"com.apple.product-type.application\""` string by trimming `\"` symbols?

**Comment by alphatroya** [comment]
> I agree, good point! The last comment - could we format `"\"com.apple.product-type.application\""` string by trimming `\"` symbols?

I think that solution will be enought

**Comment by artemnovichkov** [comment]
Thank you!

**Pull request merged**